docs/using-hive.md

```diff
@@ -157,6 +157,102 @@ Credentials will be read from `~/.gcp/osServiceAccount.json`, this can be create
 bin/hiveutil create-cluster --base-domain=mydomain.example.com --cloud=gcp mycluster
 ```
 
+#### Create Cluster on Bare Metal
+
+Hive supports bare metal provisioning as provided by [openshift-install](https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md)
+
+At present this feature requires a separate pre-existing libvirt provisioning host to run the bootstrap node. This host will require very specific network configuration that far exceeds the scope of Hive documentation. See [Bare Metal Platform Customization](https://github.com/openshift/installer/blob/master/docs/user/metal/customize_ipi.md#using-a-remote-hypervisor) for more information.
+
+![Bare metal provisioning with one hypervisor host per cluster](hive-baremetal-hypervisor-per-cluster.png)
+
+![Bare metal provisioning with a shared hypervisor host](hive-baremetal-shared-hypervisor.png)
+
+To provision bare metal clusters with Hive:
+
+Create a `Secret` containing a [bare metal enabled InstallConfig](https://github.com/openshift/installer/blob/master/docs/user/metal/install_ipi.md#install-config). This `InstallConfig` must contain a `libvirtURI` property pointing to the provisioning host.
+
+Create a `Secret` containing the SSH private key that can connect to your libvirt provisioning host, without a passphrase.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: provisioning-host-ssh-private-key
+stringData:
+  ssh-privatekey: |-
+    -----BEGIN RSA PRIVATE KEY-----
+   REDACTED
+    -----END RSA PRIVATE KEY-----
+type: Opaque
+```
+
+Create a `ConfigMap` for manifests to inject into the installer, containing a nested `ConfigMap` for metal3 config.
+
+*NOTE*: This will no longer be required as of OpenShift 4.4+.
+
+```yaml
+apiVersion: v1
+data:
+  99_metal3-config.yaml: |
+    kind: ConfigMap
+    apiVersion: v1
+    metadata:
+      name: metal3-config
+      namespace: openshift-machine-api
+    data:
+      http_port: "6180"
+      provisioning_interface: "enp1s0"
+      provisioning_ip: "172.22.0.3/24"
+      dhcp_range: "172.22.0.10,172.22.0.100"
+      deploy_kernel_url: "http://172.22.0.3:6180/images/ironic-python-agent.kernel"
+      deploy_ramdisk_url: "http://172.22.0.3:6180/images/ironic-python-agent.initramfs"
+      ironic_endpoint: "http://172.22.0.3:6385/v1/"
+      ironic_inspector_endpoint: "http://172.22.0.3:5050/v1/"
+      cache_url: "http://192.168.111.1/images"
+      rhcos_image_url: "https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/rhcos-4.3/43.81.201911192044.0/x86_64/rhcos-43.81.201911192044.0-openstack.x86_64.qcow2.gz"
+kind: ConfigMap
+metadata:
+  name: my-baremetal-cluster-install-manifests
+```
+
+Create a `ClusterDeployment`, note the `libvirtSSHPrivateKeyRef` and `sshKnownHosts` for bare metal:
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: ClusterDeployment
+metadata:
+  name: my-baremetal-cluster
+  annotations:
+    hive.openshift.io/try-install-once: "true"
```

Comment:
Is the try-install-once annotation something necessary for a baremetal install? Or is it just a remnant of testing?

Reply:
I'm unsure on this, there is no deprovision process for bare metal anywhere, so I'm not positive retry makes sense. If it goes bad it usually hangs indefinitely. I think try once is ok to recommend for now.

Reply:
This could go either way. It probably can't hurt to take it out. I will remove for now.

Reply:
If it does not make sense to try the install more than once for bare metal, should we build that into Hive directly rather than requiring the user to add the annotation?

Reply:
I'm really not sure, if we bake it in and change our minds, it's a pretty dramatic change in behavior that could cause some problems. I'm going to keep the annotation for now and ask around for more info, we can followup if necessary.

Reply:
I agree with @staebler. If it's a known behavior, then we should not expect the end user to supply the annotations each time they want to install with bare metal.

Reply:
Retrying won't require deprovisioning, but the types of failures we see mean it's not likely to work just by trying again. Someone usually has to go fix a switch or other infrastructure setting to get it to work.

I don't know what that means about the UX of making try-install-once the default for only one platform. Maybe the place to make that decision is in ACM, rather than Hive?